src/coreclr/pal/src/libunwind/include/libunwind-arm.h

```diff
@@ -269,32 +269,43 @@ typedef struct unw_tdep_context
-  }), 0)
+#define unw_tdep_getcontext(uc) ({                                                              \
+  unw_tdep_context_t *unw_ctx = (uc);                                                           \
+  register unsigned long *r0 __asm__ ("r0");                                                    \
+  register unsigned long *unw_base __asm__ ("r1") = unw_ctx->regs;                              \
+  __asm__ __volatile__ (                                                                        \
+    "mov r0, #0\n"                                                                              \
+    "stmia %[base]!, {r0-r15}\n"                                                                \
+    "vstmia %[base], {d0-d15}\n" /* this also aligns return address to value stored by stmia */ \
+    : [r0] "=r" (r0) : [base] "r" (unw_base) : "memory");                                       \
+  (int)r0; })
 #else /* __thumb__ */
-#define unw_tdep_getcontext(uc) (({                                     \
+#define unw_tdep_getcontext(uc) ({                                      \
   unw_tdep_context_t *unw_ctx = (uc);                                   \
-  register unsigned long *unw_base __asm__ ("r0") = unw_ctx->regs;      \
+  register unsigned long *r0 __asm__ ("r0");                            \
+  register unsigned long *unw_base __asm__ ("r1") = unw_ctx->regs;      \
   __asm__ __volatile__ (                                                \
-    ".align 2\nbx pc\nnop\n.code 32\n"                                  \
-    "stmia %[base], {r0-r15}\n"                                         \
-    "adds %[base], #64\n"                                               \
+    ".align 2\n"                                                        \
+    "bx pc\n"                                                           \
+    "nop\n"                                                             \
+    ".code 32\n"                                                        \
+    "mov r0, #0\n"                                                      \
+    "stmia %[base], {r0-r14}\n"                                         \
+    "adr r0, ret%=+1\n"                                                 \
+    "stmia %[base]!, {r0}\n"                                            \
     "vstmia %[base], {d0-d15}\n"                                        \
-    "orr %[base], pc, #1\nbx %[base]\n"                                 \
-    ".code 16\n"							\
-    : [base] "+r" (unw_base) : : "memory", "cc");                       \
-  }), 0)
+    "orr r0, pc, #1\n"                                                  \
+    "bx r0\n"                                                           \
+    ".code 16\n"                                                        \
+    "mov r0, #0\n"                                                      \
+    "ret%=:\n"                                                          \
+    : [r0] "=r" (r0) : [base] "r" (unw_base) : "memory", "cc");         \
```

Comment:
@am11, base on @vtjnash feedback, can you please make the following change?
```suggestion
    : [r0] "+r" (r0) : [base] "r" (unw_base) : "memory", "cc");         \
```
And the same in the `#if` branch above.
This marks the base as modified by the code.

Reply:
I am sorry, this is not correct, the second register should be marked instead. Fixing the comment above.

Reply:
Hmm, actually it is more tricky. The input operand cannot be marked as modified and it also cannot be marked as clobbered. It needs to be a bit different. We need to put the base as an output operand marked with + to indicate it is both input and output. Modifying the suggested change.